snoop/data/analyzers/thumbnails.py

```diff
@@ -0,0 +1,375 @@
+    'application/x-starcalc',
+    'application/x-stardraw',
+    'application/x-starwriter',
+    'image/x-freehand',
+    'image/cgm',
+    'image/tif',
+    'image/tiff',
+    'image/vnd.dxf',
+    'image/emf',
+    'image/x-emf',
+    'image/x-targa',
+    'image/x-sgf',
+    'image/x-svm',
+    'image/wmf',
+    'image/x-wmf',
+    'image/x-pict',
+    'image/x-cmx',
+    'image/x-wpg',
+    'image/x-eps',
+    'image/x-met',
+    'image/x-portable-bitmap',
+    'image/x-photo-cd',
+    'image/x-pcx',
+    'image/x-portable-graymap',
+    'image/x-portable-pixmap',
+    'image/vnd.adobe.photoshop',
+    'image/x-cmu-raster',
+    'image/x-sun-raster',
+    'image/x-xbitmap',
+    'image/x-xpixmap',
+    'text/plain',
+    'text/html',
+    'text/xml',
+    'application/xml',
+    'application/javascript',
+    'application/sla',
+    'application/vnd.ms-pki.stl',
+    'application/x-navistyle',
+    'model/stl',
+    'application/wobj',
+    'application/object',
+    'model/obj',
+    'application/ply',
+    'application/pdf',
+    'application/x-videolan',
+    'video/3gpp',
+    'video/annodex',
+    'video/dl',
+    'video/dv',
+    'video/fli',
+    'video/gl',
+    'video/mpeg',
+    'video/mp2t',
+    'video/mp4',
+    'video/quicktime',
+    'video/mp4v-es',
+    'video/ogg',
+    'video/parityfec',
+    'video/pointer',
+    'video/webm',
+    'video/vnd.fvt',
+    'video/vnd.motorola.video',
+    'video/vnd.motorola.videop',
+    'video/vnd.mpegurl',
+    'video/vnd.mts',
+    'video/vnd.nokia.interleaved-multimedia',
+    'video/vnd.vivo',
+    'video/x-flv',
+    'video/x-la-asf',
+    'video/x-mng',
+    'video/x-ms-asf',
+    'video/x-ms-wm',
+    'video/x-ms-wmv',
+    'video/x-ms-wmx',
+    'video/x-ms-wvx',
+    'video/x-msvideo',
+    'video/x-sgi-movie',
+    'video/x-matroska',
+    'video/x-theora+ogg',
+    'video/x-m4v',
+}
+
+
+def can_create(blob):
+    """Checks if thumbnail generator service can process this mime type."""
+    if blob.mime_type in THUMBNAIL_MIME_TYPES:
+        return True
+
+    return False
+
+
+def call_thumbnails_service(data, size):
+    """Executes HTTP PUT request to Thumbnail service.
+
+    Args:
+        data: the file for which a thumbnail will be created.
+        size: the size for the created thumbnail (thumbnail will be size x size)
+        """
+
+    url = settings.SNOOP_THUMBNAIL_URL + f'preview/{size}x{size}'
+    log.info(url)
+
+    session = requests.Session()
+    resp = session.post(url, files={'file': data})
+    log.info(resp.status_code)
+
+    if resp.status_code == 500:
+        print(resp.text)
+        raise SnoopTaskBroken('thumbnail service returned http 500', 'thumbnail_http_500')
+
+    if (resp.status_code != 200
+            or resp.headers['Content-Type'] != 'image/jpeg'):
+        raise RuntimeError(f"Unexpected response from thumbnails-service: {resp}")
+    return resp.content
+
+
+@snoop_task('thumbnails.get_thumbnail')
+@returns_json_blob
```

Comment:
this doesn't return anything anymore, so remove the `@returns_json_blob` decorator (since it'll create a Blob with only `null` in it and set it as a result for all the function runs, wasting resources)